nix/e2e-tests/src/k8s_tests_tester.py

```
@@ -564,9 +566,13 @@ def _test_http_endpoints(self, deployment_config: dict) -> TestResult:
             # Test first 3 hashes
             for narinfo_hash in test_hashes[:3]:
                 try:
-                    # Fetch narinfo
-                    resp = requests.get(
-                        f"{base_url}/{narinfo_hash}.narinfo", timeout=HTTP_TIMEOUT
+                    # Fetch narinfo. Retry transient post-deploy 5xx/connection
+                    # errors: ncps can be up (healthz ok) yet briefly 5xx a
+                    # narinfo during warm-up/seeding.
+                    resp = get_with_retry(
+                        lambda: requests.get(
+                            f"{base_url}/{narinfo_hash}.narinfo", timeout=HTTP_TIMEOUT
```

Ruff (0.15.15)

[warning] Function definition does not bind loop variable `narinfo_hash`

(B023)